website/docs/contributing/ADRs/overarching/separation-request-response-schemas.md

```diff
@@ -0,0 +1,41 @@
+---
+title: "ADR: Separation of request and response schemas"
+---
+
+## Background
+
+During the updating of our OpenAPI documentation, we have encountered issues related to the width and strictness of our schemas for requests and responses. Currently, we are reusing the same schema for both request and response, which has led to situations where the schemas are either too wide for a response or too strict for a request. This has caused difficulties in accurately defining the expected data structures for API interactions.
+
+## Decision
+
+After careful consideration and discussion, it has been decided to separate the request and response schemas to address the challenges we have encountered. By creating distinct schemas for requests and responses, we aim to improve the flexibility and precision of our API documentation.
+
+### Advantages
+
+Separating the schemas will allow us to establish more precise and constrained response types while enabling more forgiving request types. This approach will facilitate better alignment between the expected data structures and the actual data transmitted in API interactions.
+
+The separation of request and response schemas will provide the following benefits:
+
+1. **Enhanced clarity and correctness**: With dedicated schemas for requests and responses, we can define the precise structure and constraints for each interaction. This will help prevent situations where the schemas are overly permissive or restrictive, improving the accuracy and clarity of the API documentation.
```

Comment:
For me, the correctness and maintainability is the heart of the value here. And I think this spans more than just the documentation, the unified schemas make writing code handling code much tougher and mean we're passing a lot of `undefineds` around where they aren't necessary. I'd like to see this fleshed out a little here. Personal feeling perhaps

Reply:
I concur, the correctness is my main motivation for driving this. It's been annoying to me that we can't claim that we know the shape of a response object because we risk failing to handle a request object.

Reply:
Thanks, tried expanding a bit on it here: https://github.com/Unleash/unleash/pull/3869/commits/7bbacb8e1c64cda574ae760019d284c3414a4dce